Sources/XiEditor/AppDelegate.swift

```diff
@@ -97,31 +97,7 @@ class AppDelegate: NSObject, NSApplicationDelegate {
         }
     }
 
-    lazy var defaultConfigDirectory: URL = {
-        let applicationDirectory = FileManager.default.urls(
-            for: .applicationSupportDirectory,
-            in: .userDomainMask)
-            .first!
-            .appendingPathComponent("XiEditor")
-
-        // create application support directory and copy preferences file on first run
-        if !FileManager.default.fileExists(atPath: applicationDirectory.path) {
-            do {
-
-                try FileManager.default.createDirectory(at: applicationDirectory,
-                                                        withIntermediateDirectories: true,
-                                                        attributes: nil)
-                let preferencesPath = applicationDirectory.appendingPathComponent(PREFERENCES_FILE_NAME)
-                let defaultConfigPath = Bundle.main.url(forResource: "client_example", withExtension: "toml")
-                try FileManager.default.copyItem(at: defaultConfigPath!, to: preferencesPath)
-
-
-            } catch let err  {
-                fatalError("Failed to create application support directory \(applicationDirectory.path). \(err)")
-            }
-        }
-        return applicationDirectory
-    }()
+    var defaultConfigDirectory: URL!
```

Comment:
This smells bad. This directory never changes, and we _do_ know it at compile time, even if it doesn't exist; so maybe this should just be,

```swift
lazy var defaultConfigDirectory: URL = {
    return FileManager.default.urls(		
        for: .applicationSupportDirectory,		
         in: .userDomainMask)		
         .first!		
         .appendingPathComponent("XiEditor")
}()
```
?

This means we can also use `self.defaultConfigDirectory` in `setupConfigDirectory`.


Edit:

to clarify, one of the reasons I don't like this right now is that it's hard to reason about; we have no idea when or where this gets set, and it turns out that it gets set as a side-effect of another function.